test/unit/test_type.c

```diff
@@ -1705,6 +1705,146 @@ bool test_path_by_offset_typedef(void) {
 	mu_end;
 }
 
+bool test_offset_by_path_struct(void) {
+	RzTypeDB *typedb = rz_type_db_new();
+	const char *types_dir = TEST_BUILD_TYPES_DIR;
+	rz_type_db_init(typedb, types_dir, "x86", 64, "linux");
+
+	// simple
+
+	// eprintf("=============== SIMPLE ===============\n");
+	char *error_msg = NULL;
+	RzType *ttype = rz_type_parse_string_single(typedb->parser, "struct Hello { int32_t a; uint32_t b; };", &error_msg);
+	mu_assert_notnull(ttype, "type parse successful");
+	mu_assert_eq(ttype->kind, RZ_TYPE_KIND_IDENTIFIER, "parsed type");
+	mu_assert_streq(ttype->identifier.name, "Hello", "parsed type");
+
+	/**
+	 * Set offsets manually.
+	 * Since parse_struct_node() in librz/type/parser/types_parser.c sets them to 0.
+	 */
+	RzBaseType *btype = rz_type_get_base_type(typedb, ttype);
+	mu_assert_notnull(btype, "btype get successful");
+	RzTypeStructMember *memb_it;
+	rz_vector_foreach(&btype->struct_data.members, memb_it) {
+		if (!strcmp(memb_it->name, "a")) {
+			memb_it->offset = 0;
+		} else if (!strcmp(memb_it->name, "b")) {
+			memb_it->offset = 4;
+		}
+	}
+
+	long long offset;
+	offset = rz_type_offset_by_path(typedb, "Hello.a");
+	mu_assert_eq(offset, 0, "offset");
+	offset = rz_type_offset_by_path(typedb, "Hello.b");
+	mu_assert_eq(offset, 32, "offset");
+
+	// recursive
+
+	// eprintf("============= RECURSIVE ==============\n");
+	ttype = rz_type_parse_string_single(typedb->parser, "union World { uint64_t ulu; Hello mulu; int32_t urshak; };", &error_msg);
+	mu_assert_notnull(ttype, "type parse successful");
+	mu_assert_eq(ttype->kind, RZ_TYPE_KIND_IDENTIFIER, "parsed type");
+	mu_assert_streq(ttype->identifier.name, "World", "parsed type");
+
+	btype = rz_type_get_base_type(typedb, ttype);
+	mu_assert_notnull(btype, "btype get successful");
+	RzTypeUnionMember *umemb_it;
+	rz_vector_foreach(&btype->struct_data.members, umemb_it) {
+		if (!strcmp(umemb_it->name, "ulu")) {
+			umemb_it->offset = 0;
+		} else if (!strcmp(umemb_it->name, "mulu")) {
+			umemb_it->offset = 0;
+		} else if (!strcmp(umemb_it->name, "urshak")) {
+			umemb_it->offset = 0;
```

Comment:
Weren't they already set all to 0 according to the previous comment?

Reply:
Hello. Yes, it is. Fixed, thanks